[14.0] connector_importer: misc fix/imp

simahawk opened the pull request:
See atomic commits for details.

simahawk committed:
connector_importer: fix non mapped key caching

If the keys in an item were different, the cache was not updated.

simahawk committed:
connector_importer: fix dyn mapper handling of tech keys

Tech keys starting w/ _ should not be discarded
otherwise you cannot use them to control certain logics
which is their main purpose.

At the end of the process they will be purged anyway by the odoo record
handler. In case you are using the mapper directly, you'll have to purge
them by yourself.

simahawk committed:
connector_importer: recognize /id keys as xmlid

It's now easier to reuse std data as all keys ending with /id
will be treated as xmlids.

simahawk committed:
connector_importer: support no time for datetime format

If there's not time specified, fallback to 00:00:00.

simahawk committed:
connector_importer: fix warning on logger.warn

rrebollo commented on connector_importer/tests/test_mapper.py:
```diff
@@ -149,6 +149,22 @@ def test_dynamic_mapper_values_with_prefix(self):
         }
         self.assertEqual(mapper.dynamic_fields(rec), expected)
 
+    def test_dynamic_mapper_values_with_slash_id(self):
```
a test on the _non_mapped_keys_cache new behavior would also be great.

rrebollo reviewed (approved):
Code Review. LGTM! When merged we need to do some forward ports.

simahawk commented:
/ocabot merge minor

simahawk committed:
connector_importer: purge values on create too

Values might contain tech keys or other invalid keys.
Make sure they get purged before create too.

simahawk commented:
Broken on 14.0 too. Fix here https://github.com/OCA/connector-interfaces/pull/164

simahawk commented:
/ocabot merge minor
